### PR TITLE
IRSA pipeline support for Terraform Deployment

### PR DESCRIPTION
**Pull request by ryansteakley**
**Which issue is resolved by this Pull Request:**
Resolves #

**Description of your changes:**
Modifies, terraform files to support pipeline irsa

**Testing:**
- [ ] Unit tests pass
- [ ] e2e tests pass
- Details about new tests (If this PR adds a new feature)
- Details about any manual tests performed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,22 +91,31 @@ port-forward:
...
+bootstrap-pipelines: verify-cluster-variables connect-to-eks-cluster
+	yq e '.cluster.name=env(CLUSTER_NAME)' -i tests/e2e/utils/pipelines/config.yaml
+	yq e '.cluster.region=env(CLUSTER_REGION)' -i tests/e2e/utils/pipelines/config.yaml
+	cd tests/e2e && PYTHONPATH=.. python3 utils/pipelines/setup_pipelines_irsa.py
 
 cleanup-ack-req: verify-cluster-variables
 	yq e '.cluster.name=env(CLUSTER_NAME)' -i tests/e2e/utils/ack_sm_controller_bootstrap/config.yaml
 	yq e '.cluster.region=env(CLUSTER_REGION)' -i tests/e2e/utils/ack_sm_controller_bootstrap/config.yaml
-	cd tests/e2e && PYTHONPATH=.. python3.8 utils/ack_sm_controller_bootstrap/cleanup_sm_controller_req.py
+	cd tests/e2e && PYTHONPATH=.. python3 utils/ack_sm_controller_bootstrap/cleanup_sm_controller_req.py
```
Comment (jsitu777) [inline Makefile:104]: could you revert these python version changes

File: awsconfigs/apps/pipeline/s3/disable-minio-server-resources.yaml [hunk 2]
```diff
@@ -17,4 +17,4 @@ apiVersion: v1
 kind: Service
 metadata:
   name: minio-service
-  namespace: kubeflow
+  namespace: kubeflow
```
Comment (jsitu777) [inline awsconfigs/apps/pipeline/s3/disable-minio-server-resources.yaml:20]: no change here

File: awsconfigs/apps/pipeline/s3/service-account.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ml-pipeline
+  namespace: kubeflow
+  annotations:
+    eks.amazonaws.com/role-arn: '{{ .Values.irsa.roleArn }}'
```
Comment (jsitu777) [inline awsconfigs/apps/pipeline/s3/service-account.yaml:7]: this is for helmify tool purpose? if yes the templates should go into the tool folder and no {{ .Values. }} here, it should only get overrides when you run the hemify script

File: charts/apps/kubeflow-pipelines/rds-s3-static/templates/ConfigMap/ml-pipeline-ui-configmap-kubeflow-ConfigMap.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+data:
+  viewer-pod-template.json: "{\n  \"spec\": {\n    \"containers\": [\n      {\n  \
```
Comment (jsitu777) [inline charts/apps/kubeflow-pipelines/rds-s3-static/templates/ConfigMap/ml-pipeline-ui-configmap-kubeflow-ConfigMap.yaml:3]: maybe reformat this by grabbing a more readable yaml file from `tools/helmify/generated_output/kustomized_output_files`. This is due to the python yaml write library

File: charts/apps/kubeflow-pipelines/s3-only-static/templates/ConfigMap/kubeflow-pipelines-profile-controller-code-kubeflow-ConfigMap.yaml [hunk 5]
```diff
@@ -0,0 +1,254 @@
+apiVersion: v1
+data:
+  sync.py: "# Copyright 2020-2021 The Kubeflow Authors\n#\n# Licensed under the Apache\
```
Comment (jsitu777) [inline charts/apps/kubeflow-pipelines/s3-only-static/templates/ConfigMap/kubeflow-pipelines-profile-controller-code-kubeflow-ConfigMap.yaml:3]: grab a more readable version from the kustomized.yaml from `tools/helmify/generated_output/kustomized_output_files`

**Review by jsitu777 — CHANGES_REQUESTED** [review]
Great work, added some comments.

File: deployments/cognito-rds-s3/terraform/cognito-rds-s3-components/main.tf [hunk 6]
```diff
@@ -62,6 +73,21 @@ module "kubeflow_secrets_manager_irsa" {
...
 }
 
+module "kubeflow_pipeline_irsa" {
+  count                             = var.use_static ? 0 : 1
+  source                            = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/irsa?ref=v4.12.1"
+  kubernetes_namespace              = kubernetes_namespace.kubeflow.metadata[0].name
+  create_kubernetes_namespace       = false
+  create_kubernetes_service_account = false
+  kubernetes_service_account        = "ml-pipeline"
+  irsa_iam_role_name                = format("%s-%s-%s-%s", "ml-pipeline", "irsa", var.addon_context.eks_cluster_id, var.addon_context.aws_region_name)
```
Comment (ananth102) [inline deployments/cognito-rds-s3/terraform/cognito-rds-s3-components/main.tf:83]: make sure this is under 64 characters.

File: Makefile (hunk 1 above)
Comment (ryansteakley) [reply Makefile]: done

File: charts/apps/kubeflow-pipelines/rds-s3-static/templates/ConfigMap/ml-pipeline-ui-configmap-kubeflow-ConfigMap.yaml (hunk 4 above)
Comment (ryansteakley) [reply charts/apps/kubeflow-pipelines/rds-s3-static/templates/ConfigMap/ml-pipeline-ui-configmap-kubeflow-ConfigMap.yaml]: will- do when I regenearte with the alpha 7 version

File: charts/apps/kubeflow-pipelines/s3-only-static/templates/ConfigMap/kubeflow-pipelines-profile-controller-code-kubeflow-ConfigMap.yaml (hunk 5 above)
Comment (ryansteakley) [reply charts/apps/kubeflow-pipelines/s3-only-static/templates/ConfigMap/kubeflow-pipelines-profile-controller-code-kubeflow-ConfigMap.yaml]: same as previous comment

**Review by jsitu777 — APPROVED** [review]

**Review by ananth102 — APPROVED** [review]

**Pull request merged**